crates/cxx-qt-lib/src/core/qmetaobjectconnectionguard.rs

```diff
@@ -0,0 +1,53 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+#[cxx::bridge]
+mod ffi {
+    extern "Rust" {
+        #[doc(hidden)]
+        type QMetaObjectConnectionGuard;
+    }
+}
+
+/// Represents a guard to a signal-slot (or signal-functor) connection.
+///
+/// This struct can be created from a [cxx_qt::QMetaObjectConnection].
+///
+/// Note that when this struct is dropped the connection is disconnected.
+/// So to keep a connection active either hold onto the struct for the duration
+/// that the connection should be active or call `release`.
+pub struct QMetaObjectConnectionGuard {
+    inner: Option<cxx_qt::QMetaObjectConnection>,
+}
+
+impl From<cxx_qt::QMetaObjectConnection> for QMetaObjectConnectionGuard {
+    fn from(connection: cxx_qt::QMetaObjectConnection) -> Self {
+        Self {
+            inner: Some(connection),
+        }
+    }
+}
+
+impl Drop for QMetaObjectConnectionGuard {
+    /// Disconnect and deconstruct the connection
+    fn drop(&mut self) {
+        if let Some(inner) = self.inner.as_ref() {
+            inner.disconnect();
+        }
+    }
+}
+
+impl QMetaObjectConnectionGuard {
+    /// Release the connection without disconnecting
+    pub fn release(mut self) {
```

Comment:
would it make sense to have this method return the QMetaObjectConnection?
Then maybe we could get around using an `Option`?

If you wanted to release the guard but still keep the connection handle around anyway, this could be handy.
And if you don't use it it is just deconstructed as it should.

Reply:
Done :-)